### PR TITLE
[API-Borrowed] Get by ID

### PR DESCRIPTION
**Pull request by mustansirMuz**
Asana link: https://app.asana.com/0/1204411152293714/1204549582395937/f

Implemented the Get by ID endpoint of Borrowed. Librarian can get any borrowed, user can only get their own.

### CODE REVIEW

File: tests/test_borrowed_get.py [hunk 1]
```diff
@@ -51,23 +51,70 @@ def test_borrowed_get_all_with_user_token(test_db: sessionmaker) -> None:
...
+
+    response = make_request(f"/borrowed/{borrowed.id}", token)
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+
+
 # Helper function
 
 
-def make_request(token: str | None = None) -> Response:
+def make_request(endpoint, token: str | None = None) -> Response:
```
Comment (Sana-Ullah786) [inline tests/test_borrowed_get.py:108]: Not A Big issue but Type annotations for endpoint is missing here.

File: tests/test_borrowed_get.py [hunk 2]
```diff
@@ -51,23 +51,70 @@ def test_borrowed_get_all_with_user_token(test_db: sessionmaker) -> None:
...
+
+
+def test_borrowed_get_by_id_without_token(test_db: sessionmaker) -> None:
+    """
+    Tests the get borrowed by id endpoint without a token. Should return 401
+    """
+    borrowed = create_borrowed(test_db)
+
+    response = make_request(f"/borrowed/{borrowed.id}")
+    response.status_code == 401
```
Comment (SohaibkhanFolio3) [inline tests/test_borrowed_get.py:84]: Use status codes from Starlette , i guess you missed it here !! This is fine aswell .

File: tests/test_borrowed_get.py (hunk 1 above)
Comment (mustansirMuz) [reply tests/test_borrowed_get.py]: Added.

File: tests/test_borrowed_get.py (hunk 2 above)
Comment (mustansirMuz) [reply tests/test_borrowed_get.py]: Fixed it. Thanks for pointing out.

**Review by SohaibkhanFolio3 — APPROVED** [review]
LGTM

**Review by Sana-Ullah786 — APPROVED** [review]
Approved. ✨.

File: tests/test_borrowed_get.py (hunk 1 above)
Comment (Sana-Ullah786) [reply tests/test_borrowed_get.py]: Done

**Pull request merged**